server/sonar-db-dao/src/main/resources/org/sonar/db/qualitygate/QualityGateMapper.xml

```diff
@@ -44,5 +44,13 @@
     where id=#{id}
   </update>
 
+  <update id="ensureOneBuiltInQualityGate" parameterType="string">
+    UPDATE quality_gates
```

Comment:
I don't think we should blindly update quality gates when there are many built-in, as when this happen it's because someone has hacked the DB.
I would prefer either : 
- select all built-in quality gates, then update them one by one, expect Sonar way
- or simply count the number of built-in qg, and fail if result is more than one

Reply:
I think failing is a very bad idea because SonarQube won't be able to start.

But I'm not seeing any difference difference with this update and "select all built-in quality gates, then update them one by one, expect Sonar way". 
The SQL command is exactly ensuring the contract : only "Sonar way" can have the flag built-in.

Reply:
If you do one SQL to get all built-in, you better control what is done : you know there's something wrong, then you can either fail or fix the issue.
Alway blindly updating none "Sonar way" built-in quality gate seems dangerous to me, as you don't know what is done.